tests/test_cuda_bd_exchange_mover.py

```diff
@@ -132,46 +201,17 @@ def test_pair_of_waters_in_box(moves, precision, rtol, atol, seed):
 
     ref_bdem = RefBDExchangeMove(nb.potential.beta, cutoff, params, group_idxs, DEFAULT_TEMP)
 
-    assert bdem.last_log_probability() == 0.0, "First log probability expected to be zero"
-    last_conf = conf
-    accepted = 0
-    for _ in range(moves):
-        before_log_weights = ref_bdem.batch_log_weights(last_conf, box)
-        x_move, x_box = bdem.move(last_conf, box)
-        after_log_weights = ref_bdem.batch_log_weights(x_move, x_box)
-        # The box will never change
-        np.testing.assert_array_equal(box, x_box)
-        num_moved = 0
-        for mol_idxs in group_idxs:
-            if not np.all(x_move[mol_idxs] == last_conf[mol_idxs]):
-                num_moved += 1
-        if num_moved > 0 and proposals_per_move == 1:
-            accepted += 1
-            # Verify that the probabilities agree when we do accept moves
-            ref_log_prob = np.minimum(logsumexp(before_log_weights) - logsumexp(after_log_weights), 0.0)
-            ref_prob = np.exp(ref_log_prob)
-            assert np.isfinite(ref_prob) and ref_prob > 0.0
-            np.testing.assert_allclose(
-                np.exp(bdem.last_log_probability()),
-                ref_prob,
-                rtol=rtol,
-                atol=atol,
-            )
-            # The molecules should all be imaged in the home box
-            np.testing.assert_allclose(image_frame(group_idxs, x_move, x_box), x_move)
-        elif num_moved == 0:
-            np.testing.assert_array_equal(last_conf, x_move)
-        assert num_moved <= 1, "More than one mol moved, something is wrong"
-        last_conf = x_move
-    # All moves are accepted, however the two waters could clash do to a proposal.
-    assert accepted == moves
+    verify_bias_deletion_moves(
+        group_idxs, bdem, ref_bdem, conf, box, proposals_per_move * moves, proposals_per_move, rtol, atol
+    )
 
 
 @pytest.mark.parametrize("precision", [np.float32])
 @pytest.mark.parametrize("seed", [2023])
 def test_bias_deletion_bulk_water_with_context(precision, seed):
     ff = Forcefield.load_default()
     system, conf, box, _ = builders.build_water_system(4.0, ff.water_ff)
+    box += np.diag([0.1, 0.1, 0.1])
```

Comment:
A lot of the systems weren't actually stable due to not adding the box padding. Added force assertions to ensure the system is reasonable.